pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -119,15 +119,18 @@ func (c *ClientCache) clientConfigForVersion(requiredVersion *schema.GroupVersio
 		return copyConfig(c.noVersionConfig), nil
 	}
 
+	// TODO this isn't what we want.  Each clientset should be setting defaults as it sees fit.
+	oldclient.SetKubernetesDefaults(&config)
+	if requiredVersion == nil {
+		requiredVersion = config.GroupVersion
```

Comment:
I'm pretty sure the method above always tries to set the group version, so this will result in always having a required version below instead of allowing a non-required version.